### PR TITLE
Add support for filtering images by aspect ratio

### PR DESCRIPTION
**Pull request by alfred-reynolds**
- Adds support for only displaying images with the selected aspect ratio (Landscape, Portrait or All).
- Adds the ability to stretch the displayed image rather than adding a border.
- Adds a "--verbose" and "-v" flag and puts debug spew behind that flag
- Updates the makefile to depend on the files in src/ to make developing easier
-

### CODE REVIEW

**Comment by alfred-reynolds** [comment]
Hey @NautiluX , sorry for the multiple features in this single PR. I'm a git/github newbie so let me know if you need me to split the changes apart.

**Review by NautiluX — COMMENTED** [review]
Thanks for submitting this PR, it looks mostly good and adds useful features. I mainly commented a few formatting nits, and probably the place for filtering based on aspect should be [the path traverser](https://github.com/NautiluX/slide/blob/4dddda44e70453bc94fc33803e8d3704fd5be111/src/pathtraverser.cpp#L10) rather than the imageselector. I think it won't work in shuffle mode the way it's implemented currently.

File: src/imageselector.cpp [hunk 1]
```diff
@@ -31,9 +96,17 @@ std::string RandomImageSelector::getNextImage()
...
-    filename = pathTraverser->getImagePath(images.at(selectedImage).toStdString());
+    while (filename.empty())
+    {
+      QStringList images = pathTraverser->getImages();
+      unsigned int selectedImage = selectRandom(images);
+      filename = pathTraverser->getImagePath(images.at(selectedImage).toStdString());
+      if (!imageValidForAspect(filename))
+      {
+        filename.clear();
+      }
```
[the remark this answers is not in the document]
Comment (alfred-reynolds) [reply src/imageselector.cpp:107]: pathTraverser->getImages() gets called EVERY time you pick a new picture in random mode and the imageValidForAspect() function needs to open the image file. I think keeping pathTraverser->getImages() being called often is great as it means we dynamically pickup new files (or remove deleted ones).
 I've gone and implemented the aspect check for the other 2 sort functions. The code added is actually similar to if we did it in path traverser due to our multiple path and sort methods.

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -164,13 +184,19 @@ void MainWindow::setOverlay(Overlay* o)
...
 
+void MainWindow::setAspect(char aspect)
+{
+  _aspect = aspect;
+}
+
 QPixmap MainWindow::getBlurredBackground(const QPixmap& originalSize, const QPixmap& scaled)
 {
-    if (scaled.width() < width()) {
+    if (scaled.width() < width() || _aspect == 'l') {
```
[the remark this answers is not in the document]
Comment (alfred-reynolds) [reply src/mainwindow.cpp]: See the change in  e238719 (part of this PR), it is needed to make sure the background QPixmap is the right size.

**Pull request merged**